lib/Github/Api/GitData/References.php

```diff
@@ -25,6 +25,20 @@ public function all($username, $repository)
         return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/refs');
     }
 
+    /**
+     * Get all matching references for the supplied reference name
+     *
+     * @param string $username
+     * @param string $repository
+     * @param string $reference
+     *
+     * @return array
+     */
+    public function matching(string $username, string $repository, string $reference): array
+    {
+        return $this->get('/repos/'.rawurlencode($username).'/'.rawurlencode($repository).'/git/matching-refs/'.rawurlencode($reference));
```

Comment:
@nickpoulos just to double check does the `rawurlencode` work with the merge of methods? Because `tags/v1.0.0` will be encoded to `tags%2Fv1.0.0`. Does the github api accept this or should we remove the `rawurlencode` now that both branch and tags are supported?

Reply:
Yea you are correct. It was very late here I did not do as thorough of a check as I should have.  The other methods that use a reference param use an existing method `$this->encodeReference()` specifically to handle these cases, which I added to my matching() method as well.